embed/etcd.go

```diff
@@ -186,6 +188,16 @@ func StartEtcd(inCfg *Config) (e *Etcd, err error) {
 	// buffer channel so goroutines on closed connections won't wait forever
 	e.errc = make(chan error, len(e.Peers)+len(e.Clients)+2*len(e.sctxs))
 
+	// newly started member ("memberInitialized==false")
+	// does not need corruption check
+	if memberInitialized {
+		if err = e.Server.CheckInitialHashKV(); err != nil {
+			// since this is before "EtcdServer.Start()"
+			// if not nil, it will block on "EtcdServer.Close()"
+			e.Server = nil
+			return e, err
```

Comment:
just return nil, err

Reply:
We still need this to close peer listeners after `startPeerListeners`

Reply:
what is the problem to return a non-nil server?

Reply:
`e.Server` is not nil, then it will block forever inside `defer` with `EtcdServer.Close`, since the server never got started.